### PR TITLE
fix(new-line-before-expect): maintain expectation indentation

### PR DESCRIPTION
**Pull request by jamesBrennan**
Currently the fix for `new-line-before-expect` will remove the indentation of the expect line.

This:
```javascript
describe("", function() {
  it("", function(){
    var a = 1
    expect(1).toBe(1)
  });
});
```

Is transformed to this:
```javascript
describe("", function() {
  it("", function(){
    var a = 1
    
expect(1).toBe(1)
  });
});
```

This PR updates the behavior so that the indentation is preserved, like this:
```javascript
describe("", function() {
  it("", function(){
    var a = 1
    
    expect(1).toBe(1)
  });
});
```

### CODE REVIEW

File: lib/rules/new-line-before-expect.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ var blockRegexp = /^((f|x)?(it|describe))$/
 module.exports = function (context) {
   var suiteDepth = 0
   var lastExpectNode
+  var lineWithWhitespace
+  var sourceCode = context.getSourceCode()
```
Comment (DianaSuvorova) [inline lib/rules/new-line-before-expect.js]: Can we move 2 new vars closer to where it is being used. It might look like we declare all the vars on top, but this wasn't the intention. The suiteDepth, lastExpectNode needed to be declared outside of the return statement.

File: lib/rules/new-line-before-expect.js [hunk 2]
```diff
@@ -12,6 +12,8 @@ var blockRegexp = /^((f|x)?(it|describe))$/
 module.exports = function (context) {
   var suiteDepth = 0
   var lastExpectNode
+  var lineWithWhitespace
```
Comment (DianaSuvorova) [inline lib/rules/new-line-before-expect.js]: can we name this var to something like: `textBeforeExpect` or something like that. Because it is not only whitespace. It can be whatever.

Also if possible can you add a test case when it is not only a whitespace, ie:

```
describe("", function() {
  it("", function(){
    var a = 1;
    var b = 2; 
    expect(1).toBe(1); expect(b).toBe(2);
  });
});
```

Comment (jamesBrennan) [reply lib/rules/new-line-before-expect.js]: No prob. Good idea on the test cases.

File: lib/rules/new-line-before-expect.js (hunk 1 above)
Comment (jamesBrennan) [reply lib/rules/new-line-before-expect.js]: sure thing.

File: lib/rules/new-line-before-expect.js (hunk 2 above)
Comment (jamesBrennan) [reply lib/rules/new-line-before-expect.js]: What is the expected output for the test case you proposed? I can't find any other examples of two expectations on the same line.

```javascript
describe("", function() {
  it("", function(){
    var a = 1;
    var b = 2; 

    expect(1).toBe(1); expect(b).toBe(2);
  });
});
```

or

```javascript
describe("", function() {
  it("", function(){
    var a = 1;
    var b = 2; 

    expect(1).toBe(1); 
    expect(b).toBe(2);
  });
});
```

Comment (DianaSuvorova) [reply lib/rules/new-line-before-expect.js]: I don't think this was thought through. Whatever makes sense to you.  I think this is rather a corner case and I just want to make sure we document the way it behaves.

Comment (jamesBrennan) [reply lib/rules/new-line-before-expect.js]: I'll go with the first option (don't break apart expectations on the same line). Although it's a bad idea to have more than one expectation on a single line enforcing that does not seem like the responsibility of this rule. It would be better handled by it's own rule - `one-expectation-per-line`, or something like that.

**Comment by jamesBrennan** [comment]
@DianaSuvorova Hold off on reviewing this. I realized I need to add another test or two to cover a few more edge cases.

**Comment by jamesBrennan** [comment]
@DianaSuvorova OK. I think we're good to go now.

File: lib/rules/new-line-before-expect.js [hunk 3]
```diff
@@ -51,3 +57,18 @@ module.exports = function (context) {
 function linesDelta (node1, node2) {
   return Math.abs(node1.loc.start.line - node2.loc.start.line)
 }
+
+function isFirstExpectOnLine (node, sourceCode) {
```
Comment (DianaSuvorova) [inline lib/rules/new-line-before-expect.js]: Oh, this function is great. Love it!

File: test/rules/new-line-before-expect.js [hunk 4]
```diff
@@ -49,6 +49,13 @@ eslintTester.run('new line before expect', rule, {
       '}));'
     ]),
     linesToCode([
+      'it("", helper(function() {',
+      '  var a = 1',
+      '',
+      '  expect(a).toBe(1); expect(a).not.toBe(0);',
```
Comment (DianaSuvorova) [inline test/rules/new-line-before-expect.js]: yeap, I agree this rule shouldn't care if there are multiple expects on the line.

File: test/rules/new-line-before-expect.js [hunk 5]
```diff
@@ -104,9 +112,78 @@ eslintTester.run('new line before expect', rule, {
...
+        '  var a = 1; expect(a).toBe(1); expect(a).not.toBe(0);',
+        ' });',
+        '});'
+      ]),
+      errors: [
+        {
+          message: 'No new line before expect'
+        }
+      ],
+      output: linesToCode([
```
Comment (DianaSuvorova) [inline test/rules/new-line-before-expect.js]: wow, that is really cool that you cover this case.
